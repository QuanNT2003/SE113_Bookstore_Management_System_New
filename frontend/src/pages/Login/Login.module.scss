.container {
    height: 100vh;
    background-image: url('../../assets/images/login-background.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
}
.login-form {
    height: 58%;
    width: 31%;
    background-color: #fff;
    border-radius: 10px;
    justify-content: center;
    align-items: center;
    display: flex;
    flex-direction: column;
    min-height: 388px;
}
.logo {
    height: 20px;
    width: 20px;
}
.form-logo {
    display: flex;
    align-items: center;
    justify-content: center;
}
.form-logo h1 {
    font-size: 22px;
    font-weight: 700;
}
.login-text {
    width: 70%;
    height: 10%;
    position: relative;
    margin-bottom: 4%;
}
.login-text input {
    outline: none;
    position: relative;
    width: 100%;
    height: 92%;
    border: none;
    font-size: 14px;
    font-weight: 500;
    color: #263c51;
    text-indent: 16px;
}
.input-border {
    // background-image: url('../../assets/images/linear-gradient-text.jpg');
    // background-color: #263c51;
    background: rgb(2, 0, 36);
    background: linear-gradient(
        90deg,
        rgba(2, 0, 36, 1) 0%,
        rgba(42, 79, 91, 1) 43%,
        rgba(0, 212, 255, 1) 100%
    );
    background-repeat: no-repeat;
    background-size: cover;
    position: relative;
    height: 8%;
    width: 100%;
}
.forgot-pass {
    width: 70%;
    height: 7%;
    position: relative;
    font-weight: 500;
    font-size: 14px;
    display: flex;
    justify-content: right;
}
.forgot-pass p {
    width: fit-content;
    height: fit-content;
    cursor: pointer;
}
.login-btn {
    height: 10%;
    width: 150px;
    font-size: 15px;
    font-weight: 700;
    color: #fff;
    background: rgb(2, 0, 36);
    background: linear-gradient(
        90deg,
        rgba(2, 0, 36, 1) 0%,
        rgba(42, 79, 91, 1) 43%,
        rgba(0, 212, 255, 1) 100%
    );
    border-radius: 20px;
    border: none;
    margin-top: 20px;
    cursor: pointer;
    &:hover {
        border-radius: 8px;
        transition: ease-out 0.2s;
    }
}
.login-btn .login-btn:hover {
    box-sizing: border-box;
}
@media screen and (min-width: 360px) {
    .login-form {
        min-width: 350px;
    }
}
@media screen and (min-height: 1180px) {
    .login-form {
        max-height: 30%;
    }
}

.red {
    background: red;
}
.login-message {
    color: red;
}
.changepass-content {
    margin-top: 20px;
    width: 85%;
    // background-color: #007bff;
    display: grid;
    height: 40%;
    grid-template-columns: 48% 48%;
    grid-template-rows: 50% 50%;
    grid-gap: 2%;
}
.grid-content {
    position: relative;
    display: flex;
    flex-direction: column;
}
.grid-content p {
    font-size: 14px;
    height: fit-content;
    width: 100%;
}
.text-inp {
    width: 100%;
    height: 45px;
    border-radius: 4px;
    border: 1px solid #c4ccde;
    background: #fff;
    text-indent: 10px;
    font-size: 14px;
    &:hover {
        border-radius: 4px;
        border: 1px solid #555555;
        background: #fff;
    }
    &:focus {
        border-radius: 4px;
        border: 1px solid #007bff;
        background: #fff;
    }
    outline: none;
}
.btn-cancel {
    margin-right: 20px;
}
.m-b {
    margin-bottom: 10px;
}
.renew-container {
    display: flex;
    flex-direction: column;
    // justify-content: center;
    align-items: center;
}
