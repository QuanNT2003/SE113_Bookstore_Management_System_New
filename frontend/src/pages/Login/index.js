import classNames from 'classnames/bind';
import styles from './Login.module.scss';
import logo from '../../assets/images/logo.png';
import { useState } from 'react';

const cx = classNames.bind(styles);
function Login() {
    const [email, SetEmail] = useState();

    const OnChangeEmail = (e) => {
        SetEmail(e.target.value);
    };

    const handleCheckingEmail = () => {
        console.log(email);
    };

    return (
        <div className={cx('container')}>
            <div className={cx('login-form')}>
                <div className={cx('form-logo')}>
                    <img src={logo} className={cx('logo')}></img>
                    <h1>Triple K</h1>
                </div>
                <p
                    style={{
                        fontSize: '14px',
                        fontStyle: 'normal',
                        fontWeight: '700',
                        textAlign: 'center',
                        width: '100%',
                        height: 'fit-content',
                        marginBottom: '2%',
                    }}
                >
                    Đăng nhập vào cửa hàng của bạn
                </p>
                <div className={cx('login-text')}>
                    <input
                        type="text"
                        placeholder="Số điện thoại hoặc email"
                        id="account"
                        value={email}
                        onChange={OnChangeEmail}
                    ></input>
                    <div className={cx('input-border')}></div>
                </div>
                <div className={cx('login-text')}>
                    <input
                        type="password"
                        placeholder="Mật khẩu"
                        id="password"
                    ></input>
                    <div className={cx('input-border')}></div>
                </div>
                <div className={cx('forgot-pass')}>
                    <p>Quên mật khẩu</p>
                </div>
                <button
                    className={cx('login-btn')}
                    onClick={handleCheckingEmail}
                >
                    ĐĂNG NHẬP
                </button>
            </div>
        </div>
    );
}

export default Login;
