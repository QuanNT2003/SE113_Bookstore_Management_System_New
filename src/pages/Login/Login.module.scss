.container {
    height: 100vh;
    background-image: url('../../assets/images/minimal-morning-landscape-8k-gx-scaled.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
}
.login-form {
    height: 58%;
    width: 31%;
    background-color: #fff;
    border-radius: 10px;
    justify-content: center;
    align-items: center;
    display: flex;
    flex-direction: column;
}
.logo {
    height: 20px;
    width: 20px;
}
.form-logo {
    display: flex;
    align-items: center;
    justify-content: center;
}
.form-logo h1 {
    font-size: 22px;
    font-weight: 700;
}
.login-text {
    width: 70%;
    height: 10%;
    position: relative;
    margin-bottom: 4%;
}
.login-text input {
    outline: none;
    position: relative;
    width: 100%;
    height: 92%;
    border: none;
    font-size: 14px;
    font-weight: 500;
    color: #316396;
    text-indent: 16px;
}
.input-border {
    background-image: url('../../assets/images/linear-gradient-text.jpg');
    background-repeat: no-repeat;
    background-size: cover;
    position: relative;
    height: 8%;
    width: 100%;
}
.forgot-pass {
    width: 70%;
    height: 7%;
    position: relative;
    font-weight: 500;
    font-size: 14px;
    display: flex;
    justify-content: right;
}
.forgot-pass p {
    width: fit-content;
    height: fit-content;
    cursor: pointer;
}
.login-btn {
    height: 10%;
    width: 47%;
    font-size: 15px;
    font-weight: 700;
    color: #fff;
    background: url('../../assets/images/linear-gradient-text.jpg') lightgray
        50%;
    border-radius: 20px;
    border: none;
    margin-top: 20px;
    cursor: pointer;
}
.login-btn .login-btn:hover {
    box-sizing: border-box;
}
@media screen and (min-width: 360px) {
    .login-form {
        min-width: 350px;
    }
}
@media screen and (min-height: 1180px) {
    .login-form {
        max-height: 30%;
    }
}
