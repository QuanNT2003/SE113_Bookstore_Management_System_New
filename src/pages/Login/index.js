import classNames from 'classnames/bind';
import styles from './Login.module.scss';
import logo from '../../assets/images/logo.png';

const cx = classNames.bind(styles);
function Login() {
    return (
        <div className={cx('container')}>
            <div className={cx('login-form')}>
                <div className={cx('form-logo')}>
                    <img src={logo} className={cx('logo')}></img>
                    <h1>Triple K</h1>
                </div>
                <p
                    style={{
                        fontSize: '14px',
                        fontStyle: 'normal',
                        fontWeight: '700',
                        textAlign: 'center',
                        width: '100%',
                        height: 'fit-content',
                        marginBottom: '2%',
                    }}
                >
                    Đăng nhập vào cửa hàng của bạn
                </p>
                <div className={cx('login-text')}>
                    <input
                        type="text"
                        placeholder="Số điện thoại hoặc email"
                        id="account"
                    ></input>
                    <div className={cx('input-border')}></div>
                </div>
                <div className={cx('login-text')}>
                    <input
                        type="password"
                        placeholder="Mật khẩu"
                        id="password"
                    ></input>
                    <div className={cx('input-border')}></div>
                </div>
                <div className={cx('forgot-pass')}>
                    <p>Quên mật khẩu</p>
                </div>
                <button className={cx('login-btn')}>ĐĂNG NHẬP</button>
            </div>
        </div>
    );
}

export default Login;
