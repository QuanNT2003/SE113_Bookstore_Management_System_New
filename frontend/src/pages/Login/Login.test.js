import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Login from '../Login/index'; // Điều chỉnh đường dẫn đến component Login của bạn

describe('Login Component Tests', () => {
    
    test('renders Login component', () => {
        render(React.createElement(Login));
        expect(screen.getByText('Đăng nhập vào cửa hàng của bạn')).toBeInTheDocument();
    });

    test('has input for phone number or email', () => {
        render(React.createElement(Login));
        expect(screen.getByPlaceholderText('Số điện thoại hoặc email')).toBeInTheDocument();
    });

    test('has input for password', () => {
        render(React.createElement(Login));
        expect(screen.getByPlaceholderText('Mật khẩu')).toBeInTheDocument();
    });

    test('has a login button', () => {
        render(React.createElement(Login));
        expect(screen.getByRole('button', { name: 'ĐĂNG NHẬP' })).toBeInTheDocument();
    });

    test('has a link for forgotten password', () => {
        render(React.createElement(Login));
        expect(screen.getByText('Quên mật khẩu')).toBeInTheDocument();
    });
});
