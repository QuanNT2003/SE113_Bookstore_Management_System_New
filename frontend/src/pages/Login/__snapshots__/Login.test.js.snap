// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="login-form"
    >
      <div
        class="form-logo"
      >
        <img
          class="logo"
          src="logo.png"
        />
        <h1>
          Triple K
        </h1>
      </div>
      <p
        style="font-size: 14px; font-style: normal; font-weight: 700; text-align: center; width: 100%; margin-bottom: 2%;"
      >
        Đăng nhập vào cửa hàng của bạn
      </p>
      <div
        class="login-text"
      >
        <input
          id="account"
          placeholder="Số điện thoại hoặc email"
          type="text"
        />
        <div
          class="input-border"
        />
      </div>
      <div
        class="login-text"
      >
        <input
          id="password"
          placeholder="Mật khẩu"
          type="password"
        />
        <div
          class="input-border"
        />
      </div>
      <div
        class="forgot-pass"
      >
        <p>
          Quên mật khẩu
        </p>
      </div>
      <button
        class="login-btn"
      >
        ĐĂNG NHẬP
      </button>
    </div>
  </div>
</DocumentFragment>
`;
